### PR TITLE
buildkite: skip clone for GitHub check notifications

### PR DESCRIPTION
**Pull request by v1v**
I copied the content of `.buildkite/scripts/build_pr_commit_status.sh` inside the Buildkite pipeline.
I enabled https://github.com/uber-workflow/run-without-clone-buildkite-plugin to avoid cloning for quick runs. See https://github.com/elastic/observability-robots/issues/2408#issuecomment-2456737659

The `Preparing working directory` won't do anything, hence I expect about 30 seconds faster builds.

image

### Improvements

From

image

now

image

### CODE REVIEW

**Comment by bmorelli25** [comment]
@v1v Thanks! It looks like this has broken the status reporting for the ci check though. The ci check is still showing as "expected"

**Comment by v1v** [comment]
> The ci check is still showing as "expected"

Thanks Brandon! I know now how to test this out :)

I'll fix it

**Comment by v1v** [comment]
@bmorelli25 , this is now working - took me a while until I passed my black belt with Buildkite 😭

File: .buildkite/build_pr_pipeline.yml [hunk 1]
```diff
@@ -6,7 +6,16 @@ steps:
     command: ".buildkite/scripts/cancel_running_pr.sh || true"
   - key: "build-pr-setup"
     label: "setup"
-    command: ".buildkite/scripts/build_pr_commit_status.sh pending"
+    command: |
+      curl -s -L \
+        -X POST \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer $${VAULT_GITHUB_TOKEN}" \
```
Comment (v1v) [inline .buildkite/build_pr_pipeline.yml:13]: avoid interpolation and therefore exposing credentials

Comment (bmorelli25) [reply .buildkite/build_pr_pipeline.yml:13]: This was an interesting read in Slack

File: .buildkite/build_pr_pipeline.yml [hunk 2]
```diff
@@ -21,11 +30,20 @@ steps:
   - key: "teardown"
     label: "teardown"
     command: |
-      if [ $(buildkite-agent step get "outcome" --step "build-pr") == "passed" ]; then
-        .buildkite/scripts/build_pr_commit_status.sh success
-      else
-        .buildkite/scripts/build_pr_commit_status.sh failure
+      status_state=failure
+      if [ $$(buildkite-agent step get "outcome" --step "build-pr") == "passed" ]; then
+        status_state=success
```
Comment (bmorelli25) [inline .buildkite/build_pr_pipeline.yml:35]: Just out of curiosity, is there a performance reason for changing this logic?

**Review by bmorelli25 — APPROVED** [review]
Thanks, Victor! I'll merge this at the start of my day tomorrow so that I can keep an eye on builds for a few hours to make sure everything is working as expected.

**Pull request merged**